File: 03/year_codes.rb
# r_codes.rb translates an academic year into a numeric code
# Name: Lily McAboy
# Date: 2/2/2024
# CS214 Lab 3
# ################################################################
#
# # Input:  The name of an academic year
# # Precondition: The academic year is a string with a value of freshman, 
# #   sophomore, junior or senior
# # Output: The corresponding integer code for the given academic year
#

def yearCode(year)
  if year =~ /freshman/
    1
  elsif year =~ /sophomore/
    2
  elsif year =~ /junior/
    3
  elsif year =~ /senior/
    4
  else
    0
  end
end

 if __FILE__ == $0
    print "Enter the year: "
    year = gets
    print "Numeric code is: "
    puts yearCode(year)
end

File: 03/YearCodes.java
///////////////////////////////////////////////////////////////////////
/* Controlling Behavior: Selection: Java exercise 
 * Name: Lily McAboy
 * Date: 2/2/2024
 * CS 214: Lab 03
 *
 */
///////////////////////////////////////////////////////////////////////

import java.util.Scanner;
//Public class YearCodes
public class YearCodes {
	//Creating a function yearcode that takes in a string
	public static int yearCode(String year)
	{
		if (year.equals("freshman"))
			return 1;
		else if (year.equals("sophomore"))
			return 2;
		else if (year.equals("junior"))
			return 3;
		else if (year.equals("senior"))
			return 4;
		else
			return 0;

	}

	//Main program
	public static void main(String[] args){
		//Prompt user for their year name
		System.out.print("\nEnter your year (freshman, sophomore, junior, or senior): ");
		//create a new scanner 
		Scanner keyboard = new Scanner(System.in);

		//getting the year from our user 
		String year = keyboard.nextLine();

		//Output 
		System.out.println("\nYour year is: " + yearCode(year) + "\n");
	}
}

